Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -15,9 +15,14 @@ import POSIX
 //FIXME: Incremental builds
 
 extension Command {
-    static func compile(clangModule module: ClangModule, externalModules: Set<Module>, configuration conf: Configuration, prefix: String) -> (Command, Command) {
-
-        let wd = Path.join(prefix, "\(module.c99name).build")
+    static func compile(clangModule module: ClangModule, externalModules: Set<Module>, configuration conf: Configuration, prefix: String) throws -> (Command, Command) {
+        
+        let wd = module.workingDirectory(prefix)
+        
+        if module.type == .Library {
+            try module.generateModuleMap(inDir: wd)
```

Comment:
In general, I don't think it is good that the code that decides whether or not to generate a module map is separate from the code which decides how to find that. I'm not familiar enough with the current code base to how hard that is to fix, here, but if there is a structural reason we can't compute these things in one place we should figure that out and fix it.